### PR TITLE
include '.sql' migrations (Fixes #3045)

### LINKED ISSUE #3045: Migration 021 missing after pip install
### CKAN Version if known (or site URL)
CKAN 2.5.2

### Please describe the expected behaviour

After installation of CKAN, the files 021_postgres_downgrade.sql, 021_postgres_upgrade.sql, 021_postgresql_downgrade.sql, and 021_postgresql_upgrade.sql should be in the migration/versions directory.

### Please describe the actual behaviour

They're not in the directory.
Running paster --plugin=ckan db init fails with a KeyError for migration 021.

### What steps can be taken to reproduce the issue? 

    sisyphus:~ $ virtualenv ./v
    Running virtualenv with interpreter /usr/bin/python2
    New python executable in /home/gbowland/v/bin/python2
    Also creating executable in /home/gbowland/v/bin/python
    Installing setuptools, pkg_resources, pip, wheel...done.
    sisyphus:~ $ . ./v/bin/activate
    (v) sisyphus:~ $ pip install ckan==2.5.2                                        
    Collecting ckan==2.5.2
    Installing collected packages: ckan
    Successfully installed ckan-2.5.2
    (v) sisyphus:~ $ ls v/lib/python2.7/site-packages/ckan/migration/versions/021*
    zsh: no matches found: v/lib/python2.7/site-packages/ckan/migration/versions/021*

Note that 021 is different in that it has .sql files, that may be the issue. Installing with pip via the git tag (git+git://github.com/ckan/ckan.git@ckan-2.5.2) works fine. If I do `python setup.py bdist` from the tag 'ckan-2.5.2` the resulting tarfile includes the .sql files for 021, so I'm a bit puzzled as to what happened.

### PR DESCRIPTION
**Pull request by grahame**
Fixes #3045 

### Proposed fixes:

install all migrations, include the migrations for 021 which are .sql files.

remove dead code from setup.py (the code in there made no
difference to the tarball produced by sdist.)

update MANIFEST.in to achieve what the stuff in setup.py
was trying to do.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [X ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by amercader** [comment]
@grahame can you summarize why the stuff you removed from setup.py is no longer needed? Does it actually prevent the sql files from being included?

Sorry, python packaging has always been hard to follow for me

**Comment by grahame** [comment]
@amercader: everything that was in setup.py is achieved through the MANIFEST.in file, so it was just duplication. I did a python setup.py sdist before and after my changes, and the only modification to the tarball produced was the inclusion of the 021 migration, and the documentation for migrations (which the config in setup.py was trying to include, unsuccessfully.)

**Comment by amercader** [comment]
For future reference `package_data` in setup.py is needed to support `python setup.py bdist`. Our release process uses `python setup.py sdist` so I think it's safe to remove for now.
